v4-migrator: `load` exception handling seems inconsistent

### Current Behavior

This is a follow-up issue to #6292.
First off, using the `5-snapshot` build result of #6302 does seem to prevent unbounded runtime. 👍 

Given the log messages there seems to be a side effect, though.

Things are progressing as expected first.
```
2026-06-08 12:46:10.193 INFO  o.d.v.load.LoadProgressReporter -   .. COMPONENT: still loading after 2h 15m 0s (expected 12937721 rows)
2026-06-08 12:46:15.193 INFO  o.d.v.load.LoadProgressReporter -   .. COMPONENT: still loading after 2h 15m 5s (expected 12937721 rows)
2026-06-08 12:46:20.193 INFO  o.d.v.load.LoadProgressReporter -   .. COMPONENT: still loading after 2h 15m 10s (expected 12937721 rows)
```
Then, some other jobs are reported to start running (but there is no completion message as seen for other operations).
*I suspect this is where the connection is actually terminated due to keep-alive timeout mismatches between servers.*
```
2026-06-08 12:46:24.448 INFO  o.d.v4migrator.load.LoadPhase - Finalizing load: re-enabling triggers and resetting identity sequences
2026-06-08 12:54:04.100 INFO  o.d.v4migrator.load.LoadPhase - Analyzing 55 loaded table(s)
2026-06-08 12:54:13.217 INFO  o.d.v4migrator.load.LoadPhase - Refreshing PORTFOLIOMETRICS_GLOBAL materialized view
2026-06-08 12:54:13.390 INFO  o.d.v4migrator.load.LoadPhase - Applying v5.7.0 cleanup deletes
```
Finally, things are terminated by a late exception seemingly stemming from the previous `COMPONENT` load operation.
```
org.jdbi.v3.core.statement.UnableToExecuteStatementException: org.postgresql.util.PSQLException: An I/O error occurred while sending to the backend. [statement:"INSERT INTO "COMPONENT" (
    "ID"
  , "AUTHORS"
  , "BLAKE2B_256"
  , "BLAKE2B_384"
  , "BLAKE2B_512"
  , "BLAKE3"
  , "CLASSIFIER"
  , "COPYRIGHT"
  , "CPE"
  , "DESCRIPTION"
  , "DIRECT_DEPENDENCIES"
  , "EXTENSION"
  , "EXTERNAL_REFERENCES"
  , "FILENAME"
  , "GROUP"
  , "INTERNAL"
  , "LAST_RISKSCORE"
  , "LICENSE"
  , "LICENSE_EXPRESSION"
  , "LICENSE_URL"
  , "MD5"
  , "NAME"
  , "TEXT"
  , "PARENT_COMPONENT_ID"
  , "PROJECT_ID"
  , "PUBLISHER"
  , "PURL"
  , "PURLCOORDINATES"
  , "LICENSE_ID"
  , "SCOPE"
  , "SHA1"
  , "SHA_256"
  , "SHA_384"
  , "SHA3_256"
  , "SHA3_384"
  , "SHA3_512"
  , "SHA_512"
  , "SUPPLIER"
  , "SWIDTAGID"
  , "UUID"
  , "VERSION"
)
SELECT "ID"
     , "AUTHORS"
     , "BLAKE2B_256"
     , "BLAKE2B_384"
     , "BLAKE2B_512"
     , "BLAKE3"
     , "CLASSIFIER"
     , "COPYRIGHT"
     , "CPE"
     , "DESCRIPTION"
     , "DIRECT_DEPENDENCIES"
     , "EXTENSION"
     , "EXTERNAL_REFERENCES"
     , "FILENAME"
     , "GROUP"
     , "INTERNAL"
     , "LAST_RISKSCORE"
     , "LICENSE"
[...]", arguments:{}]
        at org.jdbi.v3.core.statement.SqlStatements.handleException(SqlStatements.java:493)
        at org.jdbi.v3.core.statement.SqlStatement.internalExecute(SqlStatement.java:1824)
        at org.jdbi.v3.core.result.ResultProducers.lambda$returningUpdateCount$0(ResultProducers.java:49)
        at org.jdbi.v3.core.statement.Update.execute(Update.java:79)
        at org.jdbi.v3.core.statement.Update.execute(Update.java:58)
        at org.jdbi.v3.core.Handle.execute(Handle.java:354)
        at org.jdbi.v3.core.Handle.execute(Handle.java:367)
        at org.dependencytrack.v4migrator.load.LoadPhase.lambda$loadOne$0(LoadPhase.java:297)
        at org.jdbi.v3.core.transaction.LocalTransactionHandler$BoundLocalTransactionHandler.inTransaction(LocalTransactionHandler.java:229)
        at org.jdbi.v3.core.Handle.inTransaction(Handle.java:719)
        at org.jdbi.v3.core.Jdbi.lambda$inTransaction$0(Jdbi.java:457)
        at org.jdbi.v3.core.Jdbi.withHandle(Jdbi.java:423)
        at org.jdbi.v3.core.Jdbi.inTransaction(Jdbi.java:457)
        at org.dependencytrack.v4migrator.load.LoadPhase.loadOne(LoadPhase.java:297)
        at org.dependencytrack.v4migrator.load.LoadPhase.run(LoadPhase.java:80)
        at org.dependencytrack.v4migrator.cli.LoadCommand.execute(LoadCommand.java:37)
        at org.dependencytrack.v4migrator.cli.AbstractMigratorCommand.call(AbstractMigratorCommand.java:57)
        at org.dependencytrack.v4migrator.cli.AbstractMigratorCommand.call(AbstractMigratorCommand.java:38)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2031)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2469)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2461)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2423)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2425)
        at picocli.CommandLine.execute(CommandLine.java:2174)
        at org.dependencytrack.v4migrator.V4Migrator.main(V4Migrator.java:49)
        Suppressed: org.jdbi.v3.core.transaction.UnableToRestoreAutoCommitStateException: org.postgresql.util.PSQLException: This connection has been closed.
                at org.jdbi.v3.core.transaction.LocalTransactionHandler$BoundLocalTransactionHandler.restoreAutoCommitState(LocalTransactionHandler.java:270)
                at org.jdbi.v3.core.transaction.LocalTransactionHandler$BoundLocalTransactionHandler.rollback(LocalTransactionHandler.java:163)
                at org.jdbi.v3.core.Handle.rollback(Handle.java:565)
                at org.jdbi.v3.core.transaction.LocalTransactionHandler$BoundLocalTransactionHandler.inTransaction(LocalTransactionHandler.java:243)
                ... 18 more
        Caused by: org.postgresql.util.PSQLException: This connection has been closed.
                at org.postgresql.jdbc.PgConnection.checkClosed(PgConnection.java:1026)
                at org.postgresql.jdbc.PgConnection.setAutoCommit(PgConnection.java:955)
                at org.jdbi.v3.core.transaction.LocalTransactionHandler$BoundLocalTransactionHandler.restoreAutoCommitState(LocalTransactionHandler.java:268)
                ... 21 more
Caused by: org.postgresql.util.PSQLException: An I/O error occurred while sending to the backend.
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:458)
        at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:534)
        at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:437)
        at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:196)
        at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:182)
        at org.jdbi.v3.core.statement.SqlLoggerUtil.wrap(SqlLoggerUtil.java:32)
        at org.jdbi.v3.core.statement.SqlStatement.internalExecute(SqlStatement.java:1822)
        ... 25 more
Caused by: java.net.SocketException: Operation timed out
        at java.base/sun.nio.ch.SocketDispatcher.read0(Native Method)
        at java.base/sun.nio.ch.SocketDispatcher.read(Unknown Source)
        at java.base/sun.nio.ch.NioSocketImpl.tryRead(Unknown Source)
        at java.base/sun.nio.ch.NioSocketImpl.implRead(Unknown Source)
        at java.base/sun.nio.ch.NioSocketImpl.read(Unknown Source)
        at java.base/sun.nio.ch.NioSocketImpl$1.read(Unknown Source)
        at java.base/java.net.Socket$SocketInputStream.implRead(Unknown Source)
        at java.base/java.net.Socket$SocketInputStream.read(Unknown Source)
        at java.base/sun.security.ssl.SSLSocketInputRecord.read(Unknown Source)
        at java.base/sun.security.ssl.SSLSocketInputRecord.readHeader(Unknown Source)
        at java.base/sun.security.ssl.SSLSocketInputRecord.bytesInCompletePacket(Unknown Source)
        at java.base/sun.security.ssl.SSLSocketImpl.readApplicationRecord(Unknown Source)
        at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(Unknown Source)
        at org.postgresql.core.VisibleBufferedInputStream.readMore(VisibleBufferedInputStream.java:192)
        at org.postgresql.core.VisibleBufferedInputStream.ensureBytes(VisibleBufferedInputStream.java:159)
        at org.postgresql.core.VisibleBufferedInputStream.ensureBytes(VisibleBufferedInputStream.java:144)
        at org.postgresql.core.VisibleBufferedInputStream.read(VisibleBufferedInputStream.java:76)
        at org.postgresql.core.PGStream.receiveChar(PGStream.java:487)
        at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2343)
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:431)
        ... 31 more
```

It looks as if the exception thrown successfully causes the database operation to terminate early (due to the newly implemented logic).
However, the cancellation does not prevent further steps from running and seems to only be caught after the following steps complete.

### Steps to Reproduce

1. Attempt a migration
2. Experience a database connection error/timeout during `COMPONENT` load
3. Observe reported behavior

### Expected Behavior

`load` should terminate directly upon encountering an exception instead of continuing work that cannot succeed.

### Dependency-Track Version

5.x

### Browser

Google Chrome

### Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/main/CONTRIBUTING.md#filing-issues)
- [x] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this defect was already reported

--- a/support/v4-migrator/src/main/java/org/dependencytrack/v4migrator/load/LoadPhase.java
+++ b/support/v4-migrator/src/main/java/org/dependencytrack/v4migrator/load/LoadPhase.java
@@ -75,14 +75,11 @@ public final class LoadPhase {
         long totalRows = 0;
         int tableCount = 0;
         preLoad();
-        try {
-            for (final TableMigration t : TableRegistry.loaded()) {
-                totalRows += loadOne(t);
-                tableCount++;
-            }
-        } finally {
-            postLoad();
+        for (final TableMigration t : TableRegistry.loaded()) {
+            totalRows += loadOne(t);
+            tableCount++;
         }
+        postLoad();
 
         if (dropStagingAfter) {
             LOGGER.info("Dropping staging schema (--drop-staging set).");

